packages/office-addin-dev-settings/src/sideload.ts

```diff
@@ -92,6 +92,40 @@ async function generateSideloadFile(app: OfficeApp, manifest: ManifestInfo): Pro
   return pathToWrite;
 }
 
+/**
+ * Create an Office document url with query params which can be opened
+ * to register an Office add-in in Office Online.
+ * @param documentUrl Office Online document url
+ * @param manifestPath Path to the manifest file for the Office Add-in.
+ * @param port Port number for the dev-server.
+ * @param isTest Indicates whether to append test query param to suppress Office Online dialogs.
+ * @returns Document url with query params appended.
+ */
+export async function generateSideloadUrl(documentUrl: string, manifestPath: string, port: number, isTest:boolean = false): Promise<string> {
+  const regEx: RegExp = /^.*[\\\/]/;
+  const devServerPortQueryParam = "&wdaddindevserverport=";
+  const manifestFileQueryParam = "&wdaddinmanifestfile=";
+  const manifestGuidQueryParam = "&wdaddinmanifestguid=";
+  const testQueryParam = "&wdaddintest=true";
+  const manifest = await readManifestFile(manifestPath);
+  const appsInManifest: OfficeApp[] = getOfficeAppsForManifestHosts(manifest.hosts);
+  appsInManifest.forEach(element => {
+    if (element === OfficeApp.Outlook || element === OfficeApp.Project) {
+      throw new Error(`${element} is not supported for sideloading to web.`);
+    }
+  });
+
+  if (!manifest.id) {
+    throw new Error("The manifest does not contain the id for the add-in.");
+  }
+  const manifestFile: string = manifestPath.replace(regEx, '');
```

Comment:
Why are you doing this?

Reply:
We need to extract just the file name from the manifestPath, so that we can add this as a query param.  This brings to mind that I should verify that this approach will work when the manifest is in a subfolder such as c:\office-addin-takpane\manifests\manifest.xml and that it also works when the path folders have spaces in them

Reply:
Do you really need the file name of the manifest? Is this just a function of what gets shown in the dialog? Would it be better to use the add-in name which you can get from the the manifest info?

Reply:
We need to name of the manifest so the WAC code knows which file to pull from to get the manifest content. I am not prepared to make changes to the WAC code at this time if that's what you are suggesting?

Reply:
e.g. the WAC code needs the file name so it can call https://127.0. 0.1<dev-server-port>/<manifest-file-name>

Reply:
OK, this is not just the file name but it really the full path of the url to the manifest without specifying that this is from localhost, and the port number is passed as a separate argument. This is a function of the fact that the dist folder contains the manifest file and webpack-dev-server serves it up. That will be just the file name part of manifest file path. It would probably be more correct to get the file path to the root of the dev server and remove that part from the path to the manifest file, however, i'm not sure how to do that.

If you just want to get the file path from the url, there are easier ways to do that. You don't need to use your own regex. There should be an easy to get the path component, or just the file name part of the path, from the url.

Reply:
looks like path.basename will work to get the file name